pipeline/config.go

```diff
@@ -240,3 +240,39 @@ func (p Params) buildAsCustom(context plugin.BuildContext, namespace string) ([]
 
 	return config.Pipeline, nil
 }
+
+func (p *Params) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var m map[interface{}]interface{}
+	err := unmarshal(&m)
+	if err != nil {
+		return err
+	}
+
+	*p = Params(cleanMap(m))
+	return nil
+}
+
+func cleanMap(m map[interface{}]interface{}) map[string]interface{} {
+	clean := make(map[string]interface{}, len(m))
+	for k, v := range m {
+		clean[fmt.Sprintf("%v", k)] = cleanValue(v)
+	}
+	return clean
+}
+
+func cleanValue(v interface{}) interface{} {
+	switch v := v.(type) {
+	case string, bool, int, int64, int32, float32, float64:
```

Comment:
Do we want to support nil as a possible value?

Reply:
I don't see any reason not to